konnectivity-client/pkg/client/client_test.go

```diff
@@ -151,6 +152,15 @@ func TestClose(t *testing.T) {
 	}
 }
 
+func TestCreateSingleUseGrpcTunnel_NoLeakOnFailure(t *testing.T) {
```

Comment:
Goroutine leak reproduced without the fix:
```
$ go test ./...
I0223 02:03:46.766980   27663 client.go:104] "stream read failure"
I0223 02:03:46.767325   27663 client.go:104] "stream read failure"
--- FAIL: TestCreateSingleUseGrpcTunnel_NoLeakOnFailure (0.45s)
    leaks.go:78: found unexpected goroutines:
        [Goroutine 25 in state select, with google.golang.org/grpc.(*ccBalancerWrapper).watcher on top of the stack:
        goroutine 25 [select]:
        google.golang.org/grpc.(*ccBalancerWrapper).watcher(0xc0001ae4c0)
                /usr/local/google/home/andrewsy/gke-internal-kubernetes/pkg/mod/google.golang.org/grpc@v1.27.1/balancer_conn_wrappers.go:69 +0x8f
        created by google.golang.org/grpc.newCCBalancerWrapper
                /usr/local/google/home/andrewsy/gke-internal-kubernetes/pkg/mod/google.golang.org/grpc@v1.27.1/balancer_conn_wrappers.go:60 +0x1ca
        
         Goroutine 26 in state select, with google.golang.org/grpc.(*addrConn).resetTransport on top of the stack:
        goroutine 26 [select]:
        google.golang.org/grpc.(*addrConn).resetTransport(0xc0001fe000)
                /usr/local/google/home/andrewsy/gke-internal-kubernetes/pkg/mod/google.golang.org/grpc@v1.27.1/clientconn.go:1149 +0x5f7
        created by google.golang.org/grpc.(*addrConn).connect
                /usr/local/google/home/andrewsy/gke-internal-kubernetes/pkg/mod/google.golang.org/grpc@v1.27.1/clientconn.go:815 +0x145
        ]
FAIL
FAIL    sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client      0.496s
FAIL

```

Reply:
^ consistent with pprof data collected here https://github.com/kubernetes/kubernetes/pull/108215#discussion_r812506220

Reply:
@andrewsykim  IMHO it will be great to add this test back. We had a bunch of fixes in the konnectivity-client and each fix there means we need to backport it to k8s, which is painful and slow to rollout to customers.

IMHO, it seems nice to keep it, the deps didn't seem like so many. What do you think?

Reply:
yeah I'll look into adding it back, it'll mostly depend on how it effects deps which I can look into

Reply:
@andrewsykim it seems it might not be an issue with go 1.17 if we just use it in tests? See the last paragraph in the selected answer: https://stackoverflow.com/questions/64071364/best-way-to-use-test-dependencies-in-go-but-prevent-export-them

Not sure if that will do the trick. Also, kubernetes 1.21 to 1.23 is using go 1.16 so, even if it does help, not right now. The other option might be to move the tests to another pkg, but it is not great either :(

Reply:
Created https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/333 to follow-up, I'll open a PR to add these back after I test the module update in k/k

Reply:
Thanks!